File: app/api/schemas/ai.py
from pydantic import BaseModel, Field


class ActorContext(BaseModel):
    actor_type: str = Field(examples=["patient"])
    actor_id: str = Field(examples=["backend-user-ref"])
    organization_id: str | None = None
    role: str = Field(examples=["patient"])


class ConsentFlags(BaseModel):
    use_patient_context: bool = True
    store_ai_trace: bool = True


class AuthorizationContext(BaseModel):
    workflow: str = Field(examples=["medical_qa"])
    consent_flags: ConsentFlags = Field(default_factory=ConsentFlags)
    context_scope: list[str] = Field(default_factory=lambda: ["age", "allergies", "current_medications"])


class CallerInfo(BaseModel):
    service: str = Field(examples=["zamda-backend"])
    environment: str = Field(examples=["production"])


class Locale(BaseModel):
    language: str = "en"
    country: str = "NG"


class PatientContext(BaseModel):
    age: int | None = None
    sex: str | None = None
    known_conditions: list[str] = Field(default_factory=list)
    allergies: list[str] = Field(default_factory=list)
    current_medications: list[str] = Field(default_factory=list)


class ConversationContext(BaseModel):
    conversation_id: str | None = None
    recent_messages: list[dict] = Field(default_factory=list)


class MedicalQAInput(BaseModel):
    question: str = Field(min_length=1, examples=["Can I take ibuprofen if I have stomach ulcers?"])
    patient_context: PatientContext = Field(default_factory=PatientContext)
    conversation_context: ConversationContext = Field(default_factory=ConversationContext)


class MedicalQARequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: MedicalQAInput


class MedicalClaim(BaseModel):
    claim: str
    citation_ids: list[str] = Field(default_factory=list)


class MedicalQAResult(BaseModel):
    answer: str
    missing_context: list[str] = Field(default_factory=list)
    follow_up_questions: list[str] = Field(default_factory=list)
    medical_claims: list[MedicalClaim] = Field(default_factory=list)


class SafetyMetadata(BaseModel):
    risk_level: str = "low"
    action: str = "answered"
    requires_escalation: bool = False
    requires_human_review: bool = False


class ConfidenceMetadata(BaseModel):
    overall: float = 0.0
    grounding: float = 0.0
    retrieval: float = 0.0


class CitationItem(BaseModel):
    citation_id: str
    text_content: str
    score: float
    source_name: str | None = None
    source_version: str | None = None
    source_trust_tier: int | None = None
    document_title: str | None = None
    section_path: str | None = None
    page_number: int | None = None


class AuditMetadata(BaseModel):
    trace_id: str | None = None
    prompt_version: str | None = None
    model_provider: str | None = None
    model_version: str | None = None


class SymptomGuidanceInput(BaseModel):
    symptoms: str = Field(min_length=1, examples=["I have chest pain and shortness of breath"])
    patient_context: PatientContext = Field(default_factory=PatientContext)


class SymptomGuidanceRequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: SymptomGuidanceInput


class InteractionMedication(BaseModel):
    name: str = Field(min_length=1, examples=["warfarin"])
    dose: str | None = None


class InteractionCheckInput(BaseModel):
    medications: list[InteractionMedication] = Field(min_length=2)
    patient_context: PatientContext = Field(default_factory=PatientContext)


class InteractionCheckRequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: InteractionCheckInput


class InteractionResult(BaseModel):
    medications: list[str]
    severity: str
    summary: str
    recommended_action: str = "consult_clinician_or_pharmacist"
    citation_ids: list[str] = Field(default_factory=list)


class InteractionCheckResult(BaseModel):
    interactions: list[InteractionResult] = Field(default_factory=list)
    unknowns: list[str] = Field(default_factory=list)
    follow_up_questions: list[str] = Field(default_factory=list)


class InteractionCheckResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "interaction_check"
    result: InteractionCheckResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class DrugInfoInput(BaseModel):
    drug_name: str = Field(min_length=1, examples=["Augmentin"])
    requested_sections: list[str] | None = Field(default=None, examples=[["uses", "warnings", "side_effects"]])
    country: str = "NG"


class DrugInfoRequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: DrugInfoInput


class NormalizedDrug(BaseModel):
    input_name: str
    generic_name: str | None = None
    match_confidence: float = 0.0


class DrugInfoResult(BaseModel):
    normalized_drug: NormalizedDrug
    sections: dict[str, str] = Field(default_factory=dict)
    follow_up_questions: list[str] = Field(default_factory=list)


class DrugInfoResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "drug_info"
    result: DrugInfoResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class SymptomGuidanceResult(BaseModel):
    answer: str
    triage_level: str = "non_urgent"
    diagnosis_provided: bool = False
    follow_up_questions: list[str] = Field(default_factory=list)


class SymptomGuidanceResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "symptom_guidance"
    result: SymptomGuidanceResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class MedicalQAResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "medical_qa"
    result: MedicalQAResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class ContraindicationMedication(BaseModel):
    name: str = Field(min_length=1, examples=["ibuprofen"])


class ContraindicationCheckInput(BaseModel):
    medications: list[ContraindicationMedication] = Field(min_length=1)
    patient_context: PatientContext = Field(default_factory=PatientContext)


class ContraindicationCheckRequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: ContraindicationCheckInput


class ContraindicationItem(BaseModel):
    medication: str
    condition: str
    severity: str = "unknown"
    reason: str
    evidence_summary: str | None = None
    citation_ids: list[str] = Field(default_factory=list)


class ContraindicationCheckResult(BaseModel):
    contraindications: list[ContraindicationItem] = Field(default_factory=list)
    missing_context: list[str] = Field(default_factory=list)
    unknowns: list[str] = Field(default_factory=list)
    follow_up_questions: list[str] = Field(default_factory=list)


class ContraindicationCheckResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "contraindication_check"
    result: ContraindicationCheckResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class DosageMedication(BaseModel):
    name: str = Field(min_length=1, examples=["amoxicillin"])
    strength: str | None = None
    instructions: str | None = None


class DosageVerifyInput(BaseModel):
    medication: DosageMedication
    patient_context: PatientContext = Field(default_factory=PatientContext)


class DosageVerifyRequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: DosageVerifyInput


class DosageResult(BaseModel):
    medication_name: str
    stated_dosage: str
    assessment: str = "unknown"
    typical_range: str | None = None
    flags: list[str] = Field(default_factory=list)
    citation_ids: list[str] = Field(default_factory=list)


class DosageVerifyResult(BaseModel):
    dosages: list[DosageResult] = Field(default_factory=list)
    missing_context: list[str] = Field(default_factory=list)
    follow_up_questions: list[str] = Field(default_factory=list)


class DosageVerifyResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "dosage_verify"
    result: DosageVerifyResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class PrescriptionExplainInput(BaseModel):
    prescription_text: str = Field(min_length=1)
    patient_context: PatientContext = Field(default_factory=PatientContext)


class PrescriptionExplainRequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: PrescriptionExplainInput


class PrescriptionSection(BaseModel):
    title: str
    content: str
    citation_ids: list[str] = Field(default_factory=list)


class PrescriptionExplainResult(BaseModel):
    summary: str
    sections: list[PrescriptionSection] = Field(default_factory=list)
    warnings: list[str] = Field(default_factory=list)
    follow_up_questions: list[str] = Field(default_factory=list)


class PrescriptionExplainResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "prescription_explain"
    result: PrescriptionExplainResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class ChatInput(BaseModel):
    message: str = Field(min_length=1, examples=["Can I take ibuprofen if I have stomach ulcers?"])
    patient_context: PatientContext = Field(default_factory=PatientContext)
    conversation_context: ConversationContext = Field(default_factory=ConversationContext)


class ChatRequest(BaseModel):
    request_id: str | None = None
    caller: CallerInfo = Field(default_factory=CallerInfo)
    actor_context: ActorContext
    authorization_context: AuthorizationContext = Field(default_factory=AuthorizationContext)
    locale: Locale = Field(default_factory=Locale)
    input: ChatInput


class ChatResult(BaseModel):
    intent: str
    confidence: float
    answer: str


class ChatResponse(BaseModel):
    request_id: str | None = None
    status: str = "success"
    workflow: str = "chat"
    result: ChatResult | None = None
    safety: SafetyMetadata = Field(default_factory=SafetyMetadata)
    citations: list[CitationItem] = Field(default_factory=list)
    confidence: ConfidenceMetadata = Field(default_factory=ConfidenceMetadata)
    audit: AuditMetadata = Field(default_factory=AuditMetadata)


class ErrorDetail(BaseModel):
    code: str
    message: str
    retryable: bool = False
    details: dict | None = None


class ErrorResponse(BaseModel):
    request_id: str | None = None
    status: str = "error"
    error: ErrorDetail
    safety: SafetyMetadata | None = None

File: types/api/ai.ts
// ──────────────────────────────────────────────
// Shared nested types
// ──────────────────────────────────────────────

export interface ActorContext {
  actor_type: string;
  actor_id: string;
  organization_id: string | null;
  role: string;
}

export interface ConsentFlags {
  use_patient_context: boolean;
  store_ai_trace: boolean;
}

export interface AuthorizationContext {
  workflow: string;
  consent_flags: ConsentFlags;
  context_scope: string[];
}

export interface CallerInfo {
  service: string;
  environment: string;
}

export interface Locale {
  language: string;
  country: string;
}

export interface PatientContext {
  age: number | null;
  sex: string | null;
  known_conditions: string[];
  allergies: string[];
  current_medications: string[];
}

export interface ConversationContext {
  conversation_id: string | null;
  recent_messages: Record<string, unknown>[];
}

export interface SafetyMetadata {
  risk_level: string;
  action: string;
  requires_escalation: boolean;
  requires_human_review: boolean;
}

export interface ConfidenceMetadata {
  overall: number;
  grounding: number;
  retrieval: number;
}

export interface CitationItem {
  citation_id: string;
  text_content: string;
  score: number;
  source_name: string | null;
  source_version: string | null;
  source_trust_tier: number | null;
  document_title: string | null;
  section_path: string | null;
  page_number: number | null;
}

export interface AuditMetadata {
  trace_id: string | null;
  prompt_version: string | null;
  model_provider: string | null;
  model_version: string | null;
}

export interface ErrorDetail {
  code: string;
  message: string;
  retryable: boolean;
  details: Record<string, unknown> | null;
}

export interface ErrorResponse {
  request_id: string | null;
  status: "error";
  error: ErrorDetail;
  safety: SafetyMetadata | null;
}

// ──────────────────────────────────────────────
// Medical QA
// ──────────────────────────────────────────────

export interface MedicalQAInput {
  question: string;
  patient_context: PatientContext;
  conversation_context: ConversationContext;
}

export interface MedicalQARequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: MedicalQAInput;
}

export interface MedicalClaim {
  claim: string;
  citation_ids: string[];
}

export interface MedicalQAResult {
  answer: string;
  missing_context: string[];
  follow_up_questions: string[];
  medical_claims: MedicalClaim[];
}

export interface MedicalQAResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: MedicalQAResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Symptom Guidance
// ──────────────────────────────────────────────

export interface SymptomGuidanceInput {
  symptoms: string;
  patient_context: PatientContext;
}

export interface SymptomGuidanceRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: SymptomGuidanceInput;
}

export interface SymptomGuidanceResult {
  answer: string;
  triage_level: string;
  diagnosis_provided: boolean;
  follow_up_questions: string[];
}

export interface SymptomGuidanceResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: SymptomGuidanceResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Interaction Check
// ──────────────────────────────────────────────

export interface InteractionMedication {
  name: string;
  dose: string | null;
}

export interface InteractionCheckInput {
  medications: InteractionMedication[];
  patient_context: PatientContext;
}

export interface InteractionCheckRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: InteractionCheckInput;
}

export interface InteractionResult {
  medications: string[];
  severity: string;
  summary: string;
  recommended_action: string;
  citation_ids: string[];
}

export interface InteractionCheckResult {
  interactions: InteractionResult[];
  unknowns: string[];
  follow_up_questions: string[];
}

export interface InteractionCheckResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: InteractionCheckResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Drug Info
// ──────────────────────────────────────────────

export interface DrugInfoInput {
  drug_name: string;
  requested_sections?: string[] | null;
  country?: string;
}

export interface DrugInfoRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: DrugInfoInput;
}

export interface NormalizedDrug {
  input_name: string;
  generic_name: string | null;
  match_confidence: number;
}

export interface DrugInfoResult {
  normalized_drug: NormalizedDrug;
  sections: Record<string, string>;
  follow_up_questions: string[];
}

export interface DrugInfoResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: DrugInfoResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Contraindication Check
// ──────────────────────────────────────────────

export interface ContraindicationMedication {
  name: string;
}

export interface ContraindicationCheckInput {
  medications: ContraindicationMedication[];
  patient_context: PatientContext;
}

export interface ContraindicationCheckRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: ContraindicationCheckInput;
}

export interface ContraindicationItem {
  medication: string;
  condition: string;
  severity: string;
  reason: string;
  evidence_summary: string | null;
  citation_ids: string[];
}

export interface ContraindicationCheckResult {
  contraindications: ContraindicationItem[];
  missing_context: string[];
  unknowns: string[];
  follow_up_questions: string[];
}

export interface ContraindicationCheckResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: ContraindicationCheckResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Dosage Verify
// ──────────────────────────────────────────────

export interface DosageMedication {
  name: string;
  strength: string | null;
  instructions: string | null;
}

export interface DosageVerifyInput {
  medication: DosageMedication;
  patient_context: PatientContext;
}

export interface DosageVerifyRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: DosageVerifyInput;
}

export interface DosageResult {
  medication_name: string;
  stated_dosage: string;
  assessment: string;
  typical_range: string | null;
  flags: string[];
  citation_ids: string[];
}

export interface DosageVerifyResult {
  dosages: DosageResult[];
  missing_context: string[];
  follow_up_questions: string[];
}

export interface DosageVerifyResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: DosageVerifyResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Prescription Explain
// ──────────────────────────────────────────────

export interface PrescriptionExplainInput {
  prescription_text: string;
  patient_context: PatientContext;
}

export interface PrescriptionExplainRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: PrescriptionExplainInput;
}

export interface PrescriptionSection {
  title: string;
  content: string;
  citation_ids: string[];
}

export interface PrescriptionExplainResult {
  summary: string;
  sections: PrescriptionSection[];
  warnings: string[];
  follow_up_questions: string[];
}

export interface PrescriptionExplainResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: PrescriptionExplainResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Chat
// ──────────────────────────────────────────────

export interface ChatInput {
  message: string;
  patient_context: PatientContext;
  conversation_context: ConversationContext;
}

export interface ChatRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: ChatInput;
}

export interface ChatResult {
  intent: string;
  confidence: number;
  answer: string;
}

export interface ChatResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: ChatResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Prescriptions OCR
// ──────────────────────────────────────────────

export interface PrescriptionOcrCreateInput {
  image_data: string;
  image_format: string;
}

export interface PrescriptionOcrCreateRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: PrescriptionOcrCreateInput;
}

export interface PrescriptionOcrJob {
  job_id: string;
  status: string;
  created_at: string;
  completed_at: string | null;
  extracted_text: string | null;
  error: string | null;
}

export interface PrescriptionOcrCreateResult {
  job: PrescriptionOcrJob;
}

export interface PrescriptionOcrCreateResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: PrescriptionOcrCreateResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

export interface PrescriptionOcrPollResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: PrescriptionOcrJob | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Reminders Parse Schedule
// ──────────────────────────────────────────────

export interface ReminderParseInput {
  raw_text: string;
}

export interface ReminderParseRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: ReminderParseInput;
}

export interface ParsedReminder {
  medication_name: string;
  dosage: string | null;
  frequency: string;
  time_of_day: string[];
  start_date: string | null;
  notes: string | null;
}

export interface ReminderParseResult {
  reminders: ParsedReminder[];
  unparsed_text: string | null;
}

export interface ReminderParseResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: ReminderParseResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Doctor Assist
// ──────────────────────────────────────────────

export interface DoctorAssistInput {
  query: string;
  patient_context: PatientContext;
  conversation_context: ConversationContext;
}

export interface DoctorAssistRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: DoctorAssistInput;
}

export interface DoctorAssistResult {
  answer: string;
  recommendations: string[];
  differential_diagnoses: string[] | null;
  follow_up_questions: string[];
}

export interface DoctorAssistResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: DoctorAssistResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}

// ──────────────────────────────────────────────
// Pharmacy Assist
// ──────────────────────────────────────────────

export interface PharmacyAssistInput {
  query: string;
  patient_context: PatientContext;
  conversation_context: ConversationContext;
}

export interface PharmacyAssistRequest {
  request_id?: string | null;
  caller: CallerInfo;
  actor_context: ActorContext;
  authorization_context?: AuthorizationContext;
  locale?: Locale;
  input: PharmacyAssistInput;
}

export interface PharmacyAssistResult {
  answer: string;
  alternative_medications: string[] | null;
  insurance_notes: string[] | null;
  follow_up_questions: string[];
}

export interface PharmacyAssistResponse {
  request_id: string | null;
  status: string;
  workflow: string;
  result: PharmacyAssistResult | null;
  safety: SafetyMetadata;
  citations: CitationItem[];
  confidence: ConfidenceMetadata;
  audit: AuditMetadata;
}
